internal/jobcontainers/jobcontainer.go

```diff
@@ -368,12 +370,88 @@ func (c *JobContainer) shutdown(ctx context.Context) error {
 // supported property is schema2.PTStatistics.
 func (c *JobContainer) PropertiesV2(ctx context.Context, types ...hcsschema.PropertyType) (*hcsschema.Properties, error) {
-	return nil, errors.New("`PropertiesV2` call is not implemented for job containers")
+	if len(types) == 0 {
+		return nil, errors.New("no property types supplied for PropertiesV2 call")
+	}
+	if types[0] != hcsschema.PTStatistics {
+		return nil, errors.New("PTStatistics is the only supported property type for job containers")
+	}
+
+	memInfo, err := c.job.QueryMemoryStats()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to query for job containers memory information")
+	}
+
+	processorInfo, err := c.job.QueryProcessorStats()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to query for job containers processor information")
+	}
+
+	storageInfo, err := c.job.QueryStorageStats()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to query for job containers storage information")
+	}
+
+	var privateWorkingSet uint64
+	err = forEachProcessInfo(c.job, func(procInfo *winapi.SYSTEM_PROCESS_INFORMATION) {
+		privateWorkingSet += uint64(procInfo.WorkingSetPrivateSize)
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get private working set for container")
+	}
+
+	return &hcsschema.Properties{
+		Statistics: &hcsschema.Statistics{
+			Timestamp:          time.Now(),
+			Uptime100ns:        uint64(time.Since(c.startTimestamp)) / 100,
+			ContainerStartTime: c.startTimestamp,
+			Memory: &hcsschema.MemoryStats{
+				MemoryUsageCommitBytes:            memInfo.JobMemory,
+				MemoryUsageCommitPeakBytes:        memInfo.PeakJobMemoryUsed,
+				MemoryUsagePrivateWorkingSetBytes: privateWorkingSet,
+			},
+			Processor: &hcsschema.ProcessorStats{
+				RuntimeKernel100ns: uint64(processorInfo.TotalKernelTime),
+				RuntimeUser100ns:   uint64(processorInfo.TotalUserTime),
+				TotalRuntime100ns:  uint64(processorInfo.TotalKernelTime + processorInfo.TotalUserTime),
+			},
+			Storage: &hcsschema.StorageStats{
+				ReadCountNormalized:  storageInfo.IoInfo.ReadOperationCount,
+				ReadSizeBytes:        storageInfo.IoInfo.ReadTransferCount,
+				WriteCountNormalized: storageInfo.IoInfo.WriteOperationCount,
+				WriteSizeBytes:       storageInfo.IoInfo.WriteTransferCount,
+			},
+		},
+	}, nil
 }
 
 // Properties is not implemented for job containers. This is just to satisfy the cow.Container interface.
 func (c *JobContainer) Properties(ctx context.Context, types ...schema1.PropertyType) (*schema1.ContainerProperties, error) {
-	return nil, errors.New("`Properties` call is not implemented for job containers")
+	if len(types) == 0 {
+		return nil, errors.New("no property types supplied for Properties call")
+	}
+	if types[0] != schema1.PropertyTypeProcessList {
+		return nil, errors.New("ProcessList is the only supported property type for job containers")
+	}
+
+	var processList []schema1.ProcessListItem
```

Comment:
It would be nice if we could set `processList` to have the right capacity up front, since we should know how many processes we will be getting once we call `c.job.Pids`. This would require calling `Pids` twice or reorganizing the code a bit, though.

Reply:
Do you have any strong opinions on just leaving it how it is currently? Was this a "would be nice" mainly

Reply:
Don't feel strongly about it, just pointing out a potential improvement.